model/flow/sealing_segment.go

```diff
@@ -267,31 +329,63 @@ func (builder *SealingSegmentBuilder) AddBlock(block *Block) error {
 		result, err := builder.resultLookup(resultID)
 
 		if err != nil {
-			return fmt.Errorf("%w: (%x) %v", ErrSegmentResultLookup, resultID, err)
+			return fmt.Errorf("could not look up result with id=%x: %w", resultID, err)
 		}
 		builder.addExecutionResult(result)
 		builder.includedResults[resultID] = struct{}{}
 	}
 
+	// if the block commits to an unseen ProtocolStateID, add the corresponding data entry
+	err = builder.addProtocolStateEntryIfUnseen(block.Payload.ProtocolStateID)
+	if err != nil {
+		return fmt.Errorf("could not check or add protocol state entry: %w", err)
+	}
+
 	builder.blocks = append(builder.blocks, block)
 	return nil
 }
 
+// addProtocolStateEntryIfUnseen checks whether the given protocolStateID corresponds
+// to a previously unseen protocol state entry. If it does, retrieves the state entry
+// and persists it for inclusion in the resulting SealingSegment.
+// Errors expected during normal operation:
+//   - InvalidSealingSegmentError if the added block would cause an invalid resulting segment
```

Comment:
Not sure where this comes from? `protocolStateLookup` doesn't return expected errors.
Additionally confused how it can construct an invalid resulting segment